### DISCUSSION TITLE
ACInterfaceKobayashi, ACInterfaceKobayashi2 and InterfaceOrientationMaterial.

### DISCUSSION THREAD

**Discussion by KangChenRui** [idaholab/moose#20886] [Q&A Modules: Phase field] [answered]
Hi, everyone

I had a question when use the ACInterfaceKobayashi, ACInterfaceKobayashi2 and InterfaceOrientationMaterial.

this is my input file:
[input.txt](https://github.com/idaholab/moose/files/8570861/input.txt)

The error is as follows:
![error](https://user-images.githubusercontent.com/91399933/165490593-9ee8dfa2-e3ce-4aa9-a091-467f45d7a4c7.JPG)

I haven't found a solution yet, Thanks for your help.

**Comment by GiudGiud** [comment]
Hello

It seems multiple objects are declaring the same material properties.
They are probably not meant to be used together. Please read their documentation and look at the test files for each object to understand which ones should be combined. 

Guillaume

**Comment by laagesen** [comment, marked as answer]
Just to elaborate, you have multiple instances of the `InterfaceOrientationMaterial`.
```
[Materials]
  [./material]
    type = InterfaceOrientationMaterial
    op = eta1
mode_number = 6
  [../]
  [./material_2]
    type = InterfaceOrientationMaterial

    op = eta2
mode_number = 6
  [../]
```
Each instantiation is trying to create multiple material properties, including `eps`, `deps`, and so forth. You cannot have multiple material properties with the same so that is why you are getting the error.  If you want to use a multi-order parameter solidification model, you are going to need to use or possibly create a different material to do the interfacial energy anisotropy.

**Reply by KangChenRui** [reply]
Hi,Larry

According to your suggestion, I created kenerl for each order parameter.

eta1 has three kenerls, ACInterfaceKobayashi1, ACInterfaceKobayashi2 and InterfaceOrientationMaterial
eta2 has three kenerls, ACInterfaceKobayashi1_1, ACInterfaceKobayashi2_1 and InterfaceOrientationMaterial_1

It seems that this is troublesome. Do you have any better suggestions? 
Thank you for your reply

Chen Rui

**Reply by SudiptaBiswas** [reply]
Are you solving for a multiphase system? If so, you may consider using `InterfaceOrientationMultiphaseMaterial`. You can specify different property names for `kappa` and its derivatives there. If this does not work for you, you can follow these multiphase examples to modify the Kobayashi kernels and associated interface material. You don't need to duplicate codes, you can just make the property names an input parameter.

**Reply by SudiptaBiswas** [reply]
We have some multiphase examples here if you are interested. https://iopscience.iop.org/article/10.1088/1361-651X/ac46dc
